### PR TITLE
Merge dev and 3.5.1; some bug fixes

### PR DESCRIPTION
**Pull request by bjonkman**
**Feature or improvement description**
- merged OpenFAST/dev branch
- merged OpenFAST/3.5.1 branch
- fixed merge conflicts: There were many, so an additional set of eyes on these changes would be good!
- fixed some uninitialized output variables in an AeroDyn subroutine
- fixed some FlowField issues with Visual Studio build
- fixed some issues in steady-state and linearization code caused from introduction of FlowField 

**Impacted areas of the software**
All

**Additional supporting information**
Things that I have noticed still need work:
1. AeroDyn still treats `InflowOn*` variables as inputs, though they are not module inputs any more.
2. AeroDyn's `FlowField` parameter is set in the OpenFAST glue code and the ADI module (for the AeroDyn driver) instead of using an `InitInputType` variable. 
   - If InflowWind or ExternalInflow are not used in OpenFAST, this parameter is null.
   - Steady-state solver sets a constant wind speed. I've made a work-around so it avoids FlowField in this case, but when (1) is fixed, this will no longer work.
3. Linearization with InflowWind and/or AeroDyn is not working. I have updated some code so that the cases run to completion, but the results need to be looked at more carefully. At a minimum, the matrices have different dimensions. The mapping in the glue-code linearizaton code may not work either, since InflowWind's matrices don't have the same sizes they used to (most of the inputs are missing). 

**Test results, if applicable**
The linearization tests with AeroDyn fail. Other tests are now working.

### CODE REVIEW

**Review by deslaughter — APPROVED** [review]

File: modules/aerodyn/src/AeroDyn_AllBldNdOuts_IO.f90 [hunk 1]
```diff
@@ -195,20 +280,28 @@ SUBROUTINE Calc_WriteAllBldNdOutput( p, p_AD, u, m, m_AD, x, y, OtherState, Indx
...
+   real(ReKi)                                   :: R_pi(3,3,p%NumBlades)   ! Transformation from inertial to polar (same x at hub coordinate system, blade-azimuth rotated)
    real(ReKi)                                   :: psi_hub                 ! Azimuth wrt hub
-   real(ReKi)                                   :: Vind_g(3)               ! Induced velocity vector in global coordinates
-   real(ReKi)                                   :: Vind_s(3)               ! Induced velocity vector in section coordinates (AeroDyn "x-y")
- 
+   real(R8Ki), dimension(:,:,:,:), pointer      :: R_li                    ! Alias. Transformation from inertial to local-polar to airfoil (3x3xnNodesxnBlades)
+   real(R8Ki), dimension(:,:,:,:), pointer      :: R_wi                    ! Alias. Transformation from inertial to "WithoutSweepPitchTwist" or "orientationAnnulus". TODO: deprecate me.
+   integer(Intki), dimension(:)  , pointer      :: W2B                     ! Alias. Index from Wing index to Blade
+
+   ! Alias to shorten notations
```
[the remark this answers is not in the document]
Comment (bjonkman) [reply modules/aerodyn/src/AeroDyn_AllBldNdOuts_IO.f90:299]: Yes. That change is from https://github.com/OpenFAST/openfast/pull/1679, though.

File: modules/aerodyn/src/BEMTUncoupled.f90 [hunk 2]
```diff
@@ -895,92 +937,134 @@ subroutine axialInductionFromEmpiricalThrust( chi0, phi, k, F, axInd, H, momentu
...
+!!    CT(1)       = CT_1
+!!  The 3 coefficients are entirely determined from the three constraints
+subroutine secondOrderCoeffC1(a_c, s_c, CT_c, CT_1, c0, c1,c2)
+   real(R8Ki), intent(in ) :: a_c        !< value of a above which C1-continuation is sought
+   real(R8Ki), intent(in ) :: s_c        !< dCT/da(a_c),  slope at a=a_c
+   real(R8Ki), intent(in ) :: CT_c       !< CT(a_c), value at a=a_c
+   real(R8Ki), intent(in ) :: CT_1       !< CT(1), value at a=1
+   real(R8Ki), intent(out) :: c0, c1, c2 !< coefficients of the second order polynomial
+   real(R8Ki) :: denom
+   denom = (a_c**2 - 2._R8Ki*a_c + 1.0_R8Ki)
```
[the remark this answers is not in the document]
Comment (bjonkman) [reply modules/aerodyn/src/BEMTUncoupled.f90:1062]: That is a good question.... This is from some Envision code that was merged earlier (and I'm not sure is actually used in the OpenFAST AeroDyn). I am not sure if a_c can ever be 1, but going backwards in the code, it looks like there are some other numerical checks that need to happen before it even gets to this routine.

[the remark this answers is not in the document]
Comment (ebranlard) [reply modules/aerodyn/src/BEMTUncoupled.f90:1062]: a_c should never 1, its value is typically 0.4, but there are some bounds on it (never above 0.5) so we should be fine.

File: modules/aerodyn/src/AeroDyn_AllBldNdOuts_IO.f90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (ebranlard) [reply modules/aerodyn/src/AeroDyn_AllBldNdOuts_IO.f90:299]: I'm not familiar with associate statements, feel free to replace if you think it does the same but better. I'm just generally in favor of shorter notations :) I've used pointers as aliases in several places in the code (if you look for "alias", you'll likely find them, I'm often the one responsible for those...).

**Comment by deslaughter** [comment]
It probably doesn't make a significant difference, I thought there may have
been another reason for using the pointers. Just for information, associate
was introduced in Fortran 2003 and is explained on page 12 of
https://wg5-fortran.org/N1601-N1650/N1648.pdf. Its purpose is to provide
shorter notation without the overhead of declaring additional variables.

**Comment by andrew-platt** [comment]
I see some changes in AeroDyn.f90 that I can't find in either rc-3.5.1 or dev.  Are there some AD15 changes that didn't get into any of the other branches yet?

**Comment by bjonkman** [comment]
> I see some changes in AeroDyn.f90 that I can't find in either rc-3.5.1 or dev. Are there some AD15 changes that didn't get into any of the other branches yet, or are some of the differences I can't find just a result of code cleanup?

The only changes I made to AeroDyn.f90--other than fixing some spacing--are in  
- ef1267f053927dc97a2f3b848154627c4ab70333 to fix a bug and
- 91f7bfd89ca0622ac0bb77bbdf17bd572747d921 to fix uninitialized output variables. 

I think there were some changes in this branch (not in this PR) for the InflowWind/FlowField pointers that were made before my changes, though. Can you point to a specific line that you're questioning?

**Review by andrew-platt — APPROVED** [review]

**Comment by andrew-platt** [comment]
I was getting mixed up in comparing `rc-3.5.1` to `dev` to `dev-unstable-pointers` and lost track of where the `CompAeroMaps` was introduced.  I now see that was in a PR to `dev`.

Everything looks good to me!

**Comment by andrew-platt** [comment]
One item to note on this PR:  we decided to disable the `Fake5MW_AeroLin_B3_UA6` test case temporarily.  This will be revisited before the 4.0 release.

**Pull request merged**